File: app/server/api/models/airlineModel.js
const mongoose = require('mongoose');
const Schema   = mongoose.Schema;

const airlineSchema = new Schema({
    name: {
        type: String,
        required: true
    },
    code: {
        iata: {
            type: String,
            required: true
        },
        icao: {
            type: String,
            required: true
        }
    },
    url: {
        type: String,
        required: true
    },
    logo: {
        type: Schema.Types.ObjectId,
        required: true
    },
    scheduledFlights: [
        {
            type: Schema.Types.ObjectId,
            ref: 'Flight',
            required: true
        }
    ],
    flightHistory: [
        {
            type: Schema.Types.ObjectId,
            ref: 'Flight',
            required: true
        }
    ],
    images: [
        {
            src: {
                type: String,
                required: true
            }
        }
    ]
}, {timestamps : true});

module.exports = mongoose.model('Airline', airlineSchema);

File: api/rest/models/airline_model.py
from mongoengine import *


class AirlineCode(EmbeddedDocument):
    iata = StringField(required=True)
    icao = StringField(required=True)


class Airline(Document):
    name = StringField(required=True)
    short = StringField(required=True)
    code = EmbeddedDocumentField(AirlineCode)
    url = StringField(required=True)
